### ISSUE TITLE
Ability to prefix child process stdio lines with a custom string

### ISSUE THREAD

**Issue by callumlocke** [sindresorhus/execa#121]
Proposal: add a new option `prefix` (optional, string). If given, the child's stdout and stderr streams would be transformed so **each line** is prefixed with the given string.

Use case: when running several child processes in parallel and using `inherit` to pipe each child's stdio to the parent, prefixing would make it easier to see what's going on:

```js
Promise.all(jobs.map((job, i) => execa(job.cmd, job.args, {
  stdio: 'inherit',
  prefix: `[job ${i + 1} of jobs.length] `,
}))).then(() => {
  // ...
});
```

```
[job 1 of 4] blah blah
[job 3 of 4] blah blah
[job 3 of 4] blah blah blah
[job 2 of 4] blah
...
```

I think the best API would be to have a `prefix` option which would apply to both stdout and stderr streams, and also `prefixStdout` and `prefixStderr` in case you want to override them individually.

Happy to implement this if interested.

**Comment by sindresorhus** [comment]
It's not possible to intercept the stdout/stderr when you're using `inherit` (Really wish it was though).

**Comment by sindresorhus** [comment]
Relevant Node.js issue: https://github.com/nodejs/node/issues/8033

**Comment by ehmicky** [comment]
Re-opening because this feature still depends on #695.

**Comment by ehmicky** [comment]
PR at #699 implementing this.

The initial message's example would then written as:

```js
await Promise.all(jobs.map((job, i) => execa(job.cmd, job.args, {
  stdout: [
    'inherit',
    async function * (lines) {
      for await (line of lines) {
        yield `[job ${i + 1} of jobs.length] ${line}`,
      }
    }
  ],
})))
```

**Comment by ehmicky** [comment]
This feature has been just released in Execa 9.0.0. Please see (and share!) the [release post](https://medium.com/@ehmicky/execa-9-release-d0d5daaa097f) and the [changelog](https://github.com/sindresorhus/execa/releases/tag/v9.0.0).
